tests/trainer/test_trainer.py

```diff
@@ -1771,6 +1771,50 @@ def test_safe_checkpoints(self):
                     tmpdir, 5, int(self.n_epochs * 64 / self.batch_size), False, safe_weights=save_safetensors
                 )
 
+    def test_load_best_model_with_save(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            trainer = get_regression_trainer(
+                output_dir=tmpdir,
+                save_steps=5,
+                evaluation_strategy="steps",
+                eval_steps=5,
+            )
+            trainer.train()
+            # Check that we have the last known step:
+            assert os.path.exists(
+                os.path.join(tmpdir, f"checkpoint-{trainer.state.max_steps}")
+            ), f"Could not find checkpoint-{trainer.state.max_steps}"
+            # And then check the last multiple
+            last_multiple = trainer.state.max_steps - trainer.state.max_steps % 5
+            assert os.path.exists(
+                os.path.join(tmpdir, f"checkpoint-{last_multiple}")
+            ), f"Could not find checkpoint-{last_multiple}"
+
+        # Now test that using a limit works
+        with tempfile.TemporaryDirectory() as tmpdir:
+            trainer = get_regression_trainer(
+                output_dir=tmpdir,
+                save_steps=5,
+                evaluation_strategy="steps",
+                eval_steps=5,
+                load_best_model_at_end=True,
```

Comment:
What is `load_best_model_at_end` doing?

Reply:
Reworked the test to show that this will load in the model at a prior `eval` step rather than the end model